examples/lib/tour/02-composing-a-component-version/README.md

```diff
@@ -1,10 +1,299 @@
+
+First, we configure the component version provider.
+
+```go
+	provider := &compdesc.Provider{
+		Name: "acme.org",
+	}
+	fmt.Printf("  setting provider...\n")
+	err := cv.SetProvider(provider)
+	if err != nil {
+		return errors.Wrapf(err, "cannot set provider")
+	}
+```
+
+The provider is a structure with a name and some labels, we just set
+the name, here.
+
+Now, we fill the component version with content.
+First, we add some resource already located in
+some external registry. We use an OCI image here.
+A resources has some metadata, like an identity
+and a type.
+The identity is just a set of string properties,
+with at least containing the name property.
+Additional identity properties can be added via
+options.
+The type represents the logical meaning of the
+resource, here an `ociImage`.
+
+```go
+	meta, err := elements.ResourceMeta("image", resourcetypes.OCI_IMAGE)
+	if err != nil {
+		// without metadata options, there will be never be an error,
+		// bit to be complete, we just handle the error case, here.
+		return errors.Wrapf(err, "invalid resource meta")
+	}
+```
+
+In this example, we just use the name property.
+
+And most important a resource requires content.
+Content can be already present in some external
+repository. As long, as there is an access type
+for this kind of repository, we can just refer to it.
+Here, we just use an image provided by the
+OCM ecosystem.
+Supported access types can be found under
+.../pkg/contexts/ocm/accessmethods.
+
+```go
+	acc := ociartifact.New("ghcr.io/open-component-model/ocm/ocm.software/toi/installers/helminstaller/helminstaller:0.4.0")
+```
+
+Once we have both, the metadata and the content specification,
+we can now add the resource. 
+The `SetResource` methods will replace an existing resource with the same
+identity, or add the resource, if no such resource exists in the component
+version.
+
+```go
+	err = cv.SetResource(meta, acc)
+	if err != nil {
+		return errors.Wrapf(err, "cannot add access to ocmcli-image)")
+	}
+```
+
+Now, we will add a second resource, some unspecific yaml data.
+Therefore, we use the generic YAML resource type.
+In practice, you should always use a resource type describing
+the real meaning of the content, for example something like
+`kubernetesManifest`, This enables tools working with specific content
+to understand the resource set of a component version.
+
+```go
+	meta, err = elements.ResourceMeta("descriptor", resourcetypes.OCM_YAML)
+	if err != nil {
+		return errors.Wrapf(err, "invalid resource meta")
+	}
+```
+
+Besides referring to external resources, another possibility
+to add content is to directly provide the content blob. The
+used abstraction here is `blobaccess.BlobAccess`.
+Any blob content provided by an implementation of this
+interface can be added as resource.
+There are various access implementations for blobs
+taken from the local host, for example, from the filesystem,
```

Comment:
kannst du mal schauen, ob du das etwas einfacher formuliert bekommst? for example außerhalb der Klammer, dann ein weiteres for example innerhalb der Klammer. Vielleicht zwei Sätze oder das zweite Beispiel weglassen?

Reply:
done